### ISSUE TITLE
Problem with GString in DescriptorSetOptions.path and Gradle Kotlin DSL

### ISSUE THREAD

**Issue by IvBre** [google/protobuf-gradle-plugin#359]
Hello,
Maybe I'm missing something obvious, but when trying to set `DescriptorSetOptions.path` to the needed path string, the compiler is complaining about a type mismatch:
```
GString! required,
String provided
```

For example:
```
generateProtoTasks {
        all().forEach {
            it.generateDescriptorSet = true
            it.descriptorSetOptions.includeImports = true
            it.descriptorSetOptions.includeSourceInfo = true
            it.descriptorSetOptions.path = "${buildDir}/resources/main/META-INF/proto/descriptor_set.desc"
            ...
        }
}
```
since `it.descriptorSetOptions.path` is of type `GString`, by just doing above will not work, but instead I need to add it like:
```
it.descriptorSetOptions.path = GStringImpl(emptyArray(), arrayOf("${buildDir}/resources/main/META-INF/proto/descriptor_set.desc"))
```
which seems a bit too implementation specific, and should not be the correct usage of this interface IMO. 
Is there a better way of doing this or it's a drawback of using Kotlin DSL with this plugin?

Thanks in advance.

**Comment by voidzcy** [comment]
Sorry for the delayed response, since I was not quite familiar with GString stuff and did not know an answer to your question. 

I believe we are missing some Kotlin DSL interoperability considerations for [`DescriptorSetOptions.path`](https://github.com/google/protobuf-gradle-plugin/blob/8d421ce224a5f789e118bdd6d919db990751556d/src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy#L117). I can hardly find any explanations for the interoperation of passing a double-quoted string with interpolation in Kotlin DSL to Groovy online. But according to https://github.com/gradle/gradle/issues/9268, using [`Property<String>`](https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Property.html) seems to be Gradle's suggested approach.

**Comment by voidzcy** [comment]
#394 seems to fix this issue in an immediate way. I would keep this issue open for investigating `Property<String>` vs `String` solutions as well as if that change introduces unforeseen issues.

**Comment by josephglanville** [comment]
Would be good to get a release with this in it as there isn't a viable workaround other than building from master.

**Comment by voidzcy** [comment]
> Would be good to get a release with this in it as there isn't a viable workaround other than building from master.

Sure. Would plan to make a release after #414 (and probably #423) is done.

**Comment by voidzcy** [comment]
Fixed by #394 and it's working good enough.
